### DISCUSSION TITLE
Artemis patch engine explanation?

### DISCUSSION THREAD

**Discussion by goldug** [aldostools/webMAN-MOD#708] [Q&A]
Hey there.
I saw in the latest version that you've added Artemis patch engine. But how is it used?
Do we still need to use Artemis to select the cheats before starting the game, or is that built-in now? In that case, how do we choose the cheats?

**Comment by aldostools** [comment]
It can be used in 3 ways:
1- Use [Artemis 6.3.1](https://store.brewology.com/ahomebrew.php?brewid=291) or later from Brewology. It has a new option for use with webMAN MOD (In this case Artemis basically creates /dev_hdd0/tmp/art.txt and exits)
Launch the game webMAN will start the engine automatically if the file exists and Artemis is selected in /setup.ps3

2- You can put your cheats in specific folders and webMAN will copy the cheat file to /dev_hdd0/tmp/art.txt and webMAN will start the engine automatically if the file exists and Artemis is selected in /setup.ps3.

The paths available for the automatic copy of the codelist are:
&lt;path of iso> + .ncl
&lt;path of jb folder> + .ncl
/dev_hdd0/tmp/artemis/&lt;titleid>.ncl
/dev_hdd0/game/&lt;titleid>/artemis.ncl

3- You can edit the cheat file and save it from a remote device (PC, tablet, smartphone) using:
http://ip-address-of-ps3/artemis.ps3
or from PS3 browser using http://0/artemis.ps3

NOTE: Make sure that you update to the latest build currently available online, since most of these features were added recently.

**Reply by adonios77** [reply]
aldos, you are the God! but please explain a little bit to common people like us: How the "paths" and ".ncl" files should look like ? (Paths, TitleIds & what's inside .ncl)
Thank you very much

**Comment by adonios77** [comment]
_If anyone has make it work, please give an example for any specific game, let's say for "My Game [BLES12345].iso"._

_How the "paths" and "**.ncl**" files should look like ? (paths, titles & what's inside .ncl)
How to create the files and use them ?_ 

_Edited:_
0. Keep **Enabled Artemis** by checking its checkbox in **webMAN MOD Setup**. 
1. Start the **Artemis app** and check the **Options** to use **webMAN MOD  Plugin Version**.
2. **Copy the corresponding TitleID** (eg.BLES12345) (plain text) cheat codes from "PS3 Artemis Codelist Database" https://ps3.aldostools.org/codelist.html
3. **Paste** into a **new text file** and **rename/save** as **.ncl** file **where the game's '.iso' is**, (or the 'ps3 folder format' is).
eg.: for `dev_hdd0/PS3ISO/`
Place `My Game [BLES12345].ncl` in the same folder with existed `My Game [BLES12345].iso` .

**Comment by aldostools** [comment]
webMAN MOD processes the cheat files (input cheat codes):
`/dev_hdd0/tmp/art.txt`
`/dev_hdd0/tmp/art_l2.txt`
`/dev_hdd0/tmp/art_r2.txt`

The result is stored in: `/dev_hdd0/tmp/art.log`

For automatic creation of the file `/dev_hdd0/tmp/art.txt` the cheat code file when the game is mounted, the option `Artemis` must be checked in /setup.ps3 and the cheat file with the same name of the game but with extension `.ncl` must exist in the same folder of the ISO or GAME folder. 

<img width="1064" height="564" alt="image" src="https://github.com/user-attachments/assets/02524fdb-8e7c-4be9-874c-2d2e7933b5d5" />

Example:
```
/dev_hdd0/PS3ISO/My Game [BLES12345].ISO
/dev_hdd0/PS3ISO/My Game [BLES12345].ncl

/dev_ntfs1:/PS3ISO/Other Game [BLES23456].ISO
/dev_ntfs1:/PS3ISO/Other Game [BLES23456].ncl

/dev_hdd0/GAMES/Another Game [BLES34567]
/dev_hdd0/GAMES/Another Game [BLES34567].ncl
```
The cheat files can be obtained from http://ps3.aldostools.org/codelist.html

Example of a cheat file with extension .ncl:
<img width="609" height="543" alt="image" src="https://github.com/user-attachments/assets/1d84128a-65a0-438a-9404-1cb95cd47d4a" />

The cheats can be edited with the web command: `/artemis.ps3<path>`

Example:
/artemis.ps3/dev_hdd0/PS3ISO/My Game [BLES12345].ncl

**Reply by adonios77** [reply]
Thank you very much again!
So indeed, we should **Copy** the corresponding **TitleID** (eg.BLES12345) (plain text) cheat codes from "PS3 Artemis Codelist Database" https://ps3.aldostools.org/codelist.html 
**Paste** into a **new text file** and **rename/save as .ncl** file **where the game's '.iso' is**, or the 'ps3 folder format' is. 
eg.: for  `dev_hdd0/PS3ISO/`
Place `My Game [BLES12345].ncl` in the same folder with `My Game [BLES12345].iso` .
/!\ Start the **Artemis** (brew) **app** and check the **Options** to use **webMAN MOD  Plugin Version**.
